### PR TITLE
Patches for trunk

### PR DESCRIPTION
**Pull request by tbooth**
This is a mixed bag of small fixes that I think would be suitable for a point release. I've cherry-picked them from my main working branch where I'm tinkering with some more substantial changes.
If this PR remains open for the next few weeks I'll add more commits to it as I go.

### CODE REVIEW

**Comment by ewels** [comment]
Fab, thanks! Jetting off on holiday today (at the airport now), so will go through everything when I'm back if that's ok..

Phil

**Comment by ewels** [comment]
(Back in mid-Feb: super long honeymoon :D)

**Comment by tbooth** [comment]
Hi Phil,
Yes, I knew you were off but I didn't know it was a honeymoon. Congrats and have a lovely time! Hopefully I'll have a few nice, tidy and well-documented PRs lined up for you to come back to.
Cheers,
TIM

**Comment by ewels** [comment]
Hi Tim,

I'm now back from my travels and working my way through the MultiQC backlog. Is this ready to go, or would you like to add any more changes before merging?

Phil

**Comment by tbooth** [comment]
Hi,
Sadly I didn't yet get the time to add to it as I planned, but I still think these are good changes so yes please merge if you are happy.
Cheers,
TIM

File: multiqc/plots/table_object.py [hunk 1]
```diff
@@ -65,8 +63,7 @@ def __init__ (self, data, headers=[], pconfig={}):
 
             for k in keys:
                 # Unique id to avoid overwriting by other datasets
-                headers[idx][k]['rid'] = '{}_{}'.format(''.join(random.sample(letters, 4)), k)
-                headers[idx][k]['rid'] = re.sub(r'\W+', '_', headers[idx][k]['rid'])
+                headers[idx][k]['rid'] = '{}_{}'.format( id(headers[idx]), re.sub(r'\W+', '_', k) )
```
Comment (ewels) [inline multiqc/plots/table_object.py]: Nice, never used the `id()` function before!

File: multiqc/utils/config.py [hunk 2]
```diff
@@ -22,18 +22,18 @@
...
 version = pkg_resources.get_distribution("multiqc").version
-cwd = os.getcwd()
 script_path = os.path.dirname(os.path.realpath(__file__))
 git_hash = None
 git_hash_short = None
 try:
-    os.chdir(script_path)
-    git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD'], stderr=subprocess.STDOUT)
+    git_hash = subprocess.check_output( ['git', 'rev-parse', 'HEAD'],
+                                        cwd=script_path,
```
Comment (ewels) [inline multiqc/utils/config.py]: Nice! This is much cleaner. I can't find this documented for `check_output` anywhere, only `Popen`. If it works then that's great though!

**Pull request merged**